src/login.rs

```diff
@@ -380,6 +376,9 @@ impl Login {
                     imp.session.take().unwrap(),
                     true,
                 );
+
+                // Make everything invisible.
+                imp.outer_box.set_visible(false);
```

Comment:
What about making the Login widget itself invisibile/visible instead? This way we wouldn't need the outer box at all.

Reply:
Actually, that was the way I tried it first. But I noticed there is a problem: it seems a GtkStack's visible child is not allowed to be invisible. So, when pressing `Add Account` the following happens:

1. We change the visible child to the login widget by `imp.main_stack.set_visible_child(&*imp.login);` in `SessionManager:405`
2. At Some point, `AuthorizationState::WaitPhoneNumber ` will arrive and we make the login widget visible.

The problem in step 1 is that we want the GtkStack to change its visible child to a widget that has the property `visible = False` set at this point. While a new database folder is created and `WaitPhoneNumber` arrives in the background, nothing happens visually.
If you now press `Add Account` again, the GtkStack changes to the login view, as it has been made visible in the `WaitPhoneNumber` match arm in the meantime.

Here is a patch to try it out on top of this PR (just remove the ".zip" extension)
[rm-outer-box.patch.zip](https://github.com/melix99/telegrand/files/7958940/rm-outer-box.patch.zip)

Reply:
> Here is a patch to try it out on top of this PR

Unfortunately I'm out of town so I can't test right now (I'll be back in 3 days), so I trust your words. At this point we can resolve this.